### PR TITLE
Implement Icon component

### PR DESCRIPTION
**Pull request by thanhtrantruong47**
Implement Icon component

image

storybook: https://6684faab2d0c99a3e93bf145-xcqdkebaio.chromatic.com/

### CODE REVIEW

File: src/components/Icon/Icon.tsx [hunk 1]
```diff
@@ -1,11 +1,83 @@
...
+        stroke="#0D1222"
+        strokeWidth="1.2"
+        strokeMiterlimit="10"
+        strokeLinecap="round"
+        strokeLinejoin="round"
+      />
+      <circle cx="9" cy="12" r="0.8" stroke="#0D1222" strokeWidth="0.4" />
+      <circle cx="15" cy="12" r="0.8" stroke="#0D1222" strokeWidth="0.4" />
+    </svg>
+    {itemCount >= 0 && <div className={styles.badge}>{itemCount}</div>}
```
Comment (hungdo-agilityio) [inline src/components/Icon/Icon.tsx]: I think `CartIcon` should not include the `itemCount`

File: src/components/Icon/Icon.tsx [hunk 2]
```diff
@@ -1,11 +1,83 @@
...
 
-const Icon = (_: IconProps) => {
-  return <></>;
-};
+export const CartIcon = ({ itemCount }: IconCartProps) => (
+  <div className={styles.container}>
+    <svg width="24" height="24" viewBox="0 0 24 24" fill="none">
+      <path
+        d="M6.12403 9.03885C6.2568 7.97675 7.42732 7.39149 8.35666 7.92254L10.7597 9.29568C11.5282 9.73487 12.4718 9.73487 13.2403 9.29568L15.6433 7.92254C16.5727 7.39149 17.7432 7.97675 17.876 9.03885L19.2229 19.8139C19.3348 20.7092 18.6367 21.5 17.7344 21.5H6.26556C5.36332 21.5 4.66524 20.7092 4.77715 19.8139L6.12403 9.03885Z"
+        stroke="#0D1222"
```
Comment (hungdo-agilityio) [inline src/components/Icon/Icon.tsx]: help me research about `currentColor` in svg, that will help change color of the icon depend of the parent element

File: src/components/Icon/Icon.tsx [hunk 3]
```diff
@@ -1,11 +1,83 @@
...
+        strokeLinejoin="round"
+      />
+      <circle cx="9" cy="12" r="0.8" stroke="#0D1222" strokeWidth="0.4" />
+      <circle cx="15" cy="12" r="0.8" stroke="#0D1222" strokeWidth="0.4" />
+    </svg>
+    {itemCount >= 0 && <div className={styles.badge}>{itemCount}</div>}
+  </div>
+);
+
+export const LikeIcon = () => (
```
Comment (hungdo-agilityio) [inline src/components/Icon/Icon.tsx]: 1 file should contain 1 component

File: src/components/Icon/Icon.tsx (hunk 1 above)
Comment (thanhtrantruong47) [reply src/components/Icon/Icon.tsx]: Fixed

File: src/components/Icon/Icon.tsx (hunk 2 above)
Comment (thanhtrantruong47) [reply src/components/Icon/Icon.tsx]: Fixed

File: src/components/Icon/Icon.tsx (hunk 3 above)
Comment (thanhtrantruong47) [reply src/components/Icon/Icon.tsx]: Fixed

File: src/assets/images/like.svg
Comment (hungdo-agilityio) [inline src/assets/images/like.svg:1]: We already have like icon right, so this is redundant?

File: src/components/Icon/LikeIcon.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
+export const LikeIcon = () => (
```
Comment (hungdo-agilityio) [inline src/components/Icon/LikeIcon.tsx:1]: Please help to create storybook for all icon

File: src/components/Icon/LikeIcon.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+  <svg
+    width="24"
+    height="24"
+    viewBox="0 0 24 24"
+    fill="none"
+    xmlns="http://www.w3.org/2000/svg"
```
Comment (hungdo-agilityio) [inline src/components/Icon/LikeIcon.tsx]: You can remove this prop

File: src/components/Icon/LikeIcon.tsx (hunk 4 above)
Comment (thanhtrantruong47) [reply src/components/Icon/LikeIcon.tsx:1]: Done

File: src/components/Icon/LikeIcon.tsx (hunk 5 above)
Comment (thanhtrantruong47) [reply src/components/Icon/LikeIcon.tsx]: Fixed

File: src/assets/images/like.svg
Comment (thanhtrantruong47) [reply src/assets/images/like.svg:1]: Fixed

**Review by hungdo-agilityio — APPROVED** [review]

**Pull request merged**